src/NServiceBus.RabbitMQ/Configuration/ConnectionStringParser.cs

```diff
@@ -39,63 +40,51 @@ public ConnectionConfiguration Parse(string connectionString)
 
             if (ContainsKey("host"))
             {
-                ParseHosts(connectionConfiguration, this["host"] as string);
+                ParseHosts(connectionConfiguration, this["host"] as string, invalidOptionsMessage);
             }
             else
             {
-                throw new Exception("Invalid connection string. 'host' value must be supplied. e.g: \"host=myServer\"");
+                invalidOptionsMessage.AppendLine("Invalid connection string. 'host' value must be supplied. e.g: \"host=myServer\"");
             }
 
             if (ContainsKey("dequeuetimeout"))
             {
-                var message = "The 'DequeueTimeout' connection string option has been removed. Consult the documentation for further information.";
-
-                Logger.Error(message);
-
-                throw new NotSupportedException(message);
+                invalidOptionsMessage.AppendLine("The 'DequeueTimeout' connection string option has been removed. Consult the documentation for further information.");
```

Comment:
Given how `AppendLine` is being used to build up the complete error message, do we care about the unneeded line break this introduces at the end of the message? Not a huge deal, but it will look a bit strange.